src/osd/ReplicatedBackend.cc

```diff
@@ -2213,19 +2213,13 @@ bool ReplicatedBackend::handle_push_reply(pg_shard_t peer, PushReplyOp &op, Push
     } else {
       // done!
       get_parent()->on_peer_recover(
-	peer, soid, pi->recovery_info,
-	pi->stat);
-
-      object_stat_sum_t stat;
-      stat.num_bytes_recovered = pi->recovery_info.size;
-      stat.num_keys_recovered = reply->omap_entries.size();
-      stat.num_objects_recovered = 1;
+	peer, soid, pi->recovery_info);
 
       get_parent()->release_locks(pi->lock_manager);
+      object_stat_sum_t stat = pi->stat;
```

Comment:
i'd suggest move this into the `if (pushing[soid].empty()) {` branch. and better off declare it as a reference. also where `*_recovered` field are set? we don't set them in `ReplicatedBackend::prep_push()`.